tests/User/AutoLoginMiddlewareTest.php

```diff
@@ -0,0 +1,201 @@
+        $this->assertEquals($messages[0][1], 'Unable to authenticate used by cookie.');
+    }
+
+    public function testProcessInvalidAuthKey(): void
+    {
+        $this->mockDataRequest();
+        $this->mockDataCookie(["remember" => json_encode(['1', '123456', 60])]);
+        $this->mockFindIdentity();
+
+        $memory = memory_get_usage();
+        $this->loggerMock->setTraceLevel(3);
+
+        $this->autoLoginMiddlewareMock->process($this->requestMock, $this->requestHandlerMock);
+
+        $messages = $this->getInaccessibleProperty($this->loggerMock, 'messages');
+        $this->assertEquals($messages[0][1], 'Unable to authenticate used by cookie. Invalid auth key.');
+    }
+
+    public function testProcessCookieEmpty(): void
+    {
+        $this->mockDataRequest();
+        $this->mockDataCookie([]);
+        $this->mockFindIdentity();
+
+        $memory = memory_get_usage();
+        $this->loggerMock->setTraceLevel(3);
+
+        $this->autoLoginMiddlewareMock->process($this->requestMock, $this->requestHandlerMock);
+
+        $messages = $this->getInaccessibleProperty($this->loggerMock, 'messages');
+        $this->assertEquals($messages[0][1], 'Unable to authenticate used by cookie.');
+    }
+
+    public function testProcessCookieWithInvalidParams(): void
+    {
+        $this->mockDataRequest();
+        $this->mockDataCookie(["remember" => json_encode(['1', '123456', 60, "paramInvalid"])]);
+        $this->mockFindIdentity();
+
+        $memory = memory_get_usage();
+        $this->loggerMock->setTraceLevel(3);
+
+        $this->autoLoginMiddlewareMock->process($this->requestMock, $this->requestHandlerMock);
+
+        $messages = $this->getInaccessibleProperty($this->loggerMock, 'messages');
+        $this->assertEquals($messages[0][1], 'Unable to authenticate used by cookie.');
+    }
+
+    private function mockDataRequest(): void
+    {
+        $this->requestHandlerMock = $this->createMock(RequestHandlerInterface::class);
+        $this->userMock = $this->createMock(User::class);
+        $this->identityInterfaceMock = $this->createMock(IdentityInterface::class);
+
+        $this->loggerMock = $this->getMockBuilder(Logger::class)
+            ->onlyMethods(['dispatch'])
+            ->getMock();
+
+        $this->identityRepositoryInterfaceMock = $this->createMock(IdentityRepositoryInterface::class);
+        $this->autoLoginMiddlewareMock = new AutoLoginMiddleware($this->userMock, $this->identityRepositoryInterfaceMock, $this->loggerMock);
+        $this->requestMock = $this->createMock(ServerRequestInterface::class);
+    }
+
+    private function mockDataCookie(array $cookie): void
+    {
+        $this->requestMock
+            ->expects($this->any())
+            ->method('getCookieParams')
+            ->willReturn($cookie);
+    }
+
+    private function mockFindIdentity(): void
+    {
+        $this->identityRepositoryInterfaceMock
```

Comment:
You can do it directly in mockDataRequest.
Also you don't have to keep properties now

Reply:
I updated this in the last commit.

Reply:
How about also remove properties?

Reply:
With identityRepositoryInterfaceMock? What happend with this line. I used this property.